rustls/src/client/hs.rs

```diff
@@ -120,6 +120,7 @@ struct InitialState {
     dns_name: webpki::DNSName,
     transcript: HandshakeHash,
     extra_exts: Vec<ClientExtension>,
+    session_id: SessionID,
```

Comment:
I feel like this shouldn't exist in InitialState, and we should use Option instead of empty() and is_empty().

Reply:
That makes sense, I'll get this in the morning. (it's late in Pacific time)

Reply:
> I feel like this shouldn't exist in InitialState, and we should use Option instead of empty() and is_empty().

I agree that it seems like it shouldn't exist in `InitialState`.

I also agree that eventually we should use `Option<SessionID>` instead, but I don't think it has to be done in this PR.

Reply:
I suppose we could leave that until we can do the same transformation on the server side.

Reply:
Well, I did this before I saw these comments... lmk if the new diff is ok. It wasn't that hard to thread it through.

Reply:
To remove those methods I guess we'd have to do the server side, too? But I was definitely angling to have the use of those methods decimated from the client side handshake code.

Reply:
I'll resolve this conversation and move discussion down below, to the current patch.